nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/VerifyContentMAC.java

```diff
@@ -0,0 +1,187 @@
+    @WritesAttribute(attribute = VerifyContentMAC.MAC_ALGORITHM_ATTRIBUTE, description = "The MAC algorithm")})
+public class VerifyContentMAC extends AbstractProcessor {
+
+    protected static final String HMAC_SHA256 = "HmacSHA256";
+    protected static final String HMAC_SHA512 = "HmacSHA512";
+
+    protected static final PropertyDescriptor MAC_ALGORITHM = new PropertyDescriptor.Builder()
+        .name("MAC_ALGORITHM")
+        .displayName("MAC Algorithm")
+        .description("Cryptographic Hash Function")
+        .allowableValues(HMAC_SHA256, HMAC_SHA512)
+        .required(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    protected static final PropertyDescriptor SECRET_KEY = new PropertyDescriptor.Builder()
+        .name("SECRET_KEY")
+        .displayName("Secret key")
+        .description("Cryptographic key to calculate the hash")
+        .required(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .sensitive(true)
+        .build();
+
+    protected static final PropertyDescriptor MAC = new PropertyDescriptor.Builder()
+        .name("MAC")
+        .displayName("Message Authentication Code")
+        .description("The MAC to compare with the calculated value")
+        .required(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .build();
+
+    protected static final Relationship FAILURE = new Relationship.Builder()
+        .name("failure")
+        .description("Signature Verification Failed")
+        .build();
+
+    protected static final Relationship SUCCESS = new Relationship.Builder()
+        .name("success")
+        .description("Signature Verification Succeeded")
+        .build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(Arrays.asList(MAC_ALGORITHM, SECRET_KEY, MAC));
+    private static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(SUCCESS, FAILURE)));;
+    private static final Charset CHARSET = StandardCharsets.UTF_8;
+    private static final int BUFFER_SIZE = 512000;
+    private static final String HEX_FORMAT = "%02x";
+    protected static final String MAC_CALCULATED_ATTRIBUTE = "mac.calculated";
+    protected static final String MAC_ALGORITHM_ATTRIBUTE = "mac.algorithm";
+
+    private SecretKeySpec secretKeySpec;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @OnScheduled
+    public void setUp(ProcessContext context)  {
+        secretKeySpec = new SecretKeySpec(context.getProperty(SECRET_KEY).getValue().getBytes(CHARSET), HMAC_SHA256);
```

Comment:
This should be broken out to multiple lines. Also recommend adding a `Secret Key Encoding` property that defaults to `Hexadecimal`, which will make the expected input clear, and also support the possibility of another option, like `Base64`.

Reply:
Do you think it makes sense to have a raw/utf8 option as well for the secret key?

Reply:
I don't think a raw UTF-8 option makes sense because a random secret key will usually consist of bytes that are not easily represented as UTF-8 characters, so I would limit the supported options to `HEXADECIMAL` and `BASE64`.